iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java

```diff
@@ -179,12 +183,19 @@ private static boolean isSensitivePublicNetworkAccessSimplified(Tree tree) {
     return TextUtils.matchesValue(tree, "Enabled"::contains).isTrue();
   }
 
-  private static BiConsumer<CheckContext, ResourceDeclaration> checkIpRange() {
-    return (ctx, resource) -> {
-      Optional<Tree> startIpAddress = PropertyUtils.value(resource, "startIpAddress");
-      Optional<Tree> endIpAddress = PropertyUtils.value(resource, "endIpAddress");
-      IpAddressValidator validator = new IpAddressValidator((ArmTree) startIpAddress.orElse(null), (ArmTree) endIpAddress.orElse(null));
-      validator.reportIssueIfPublicIPAddress(ctx, PUBLIC_IP_ADDRESS_MESSAGE, PUBLIC_IP_ADDRESS_MESSAGE_SECONDARY_LOCATION);
+  private static <S extends ContextualMap<S, T>, T extends HasProperties & Tree> void checkIpRange(ContextualMap<S, T> resource) {
```

Comment:
I wish there was a way to write this simpler, but I don't think there is.
The only way I found was to write it this way:
`private static void checkIpRange(ContextualMap<?, ?> resource) {`
I'm not convinced it's a good idea, I let you decide on this.

Reply:
But then the generic checks are weaker. I will keep it as it is.